### PR TITLE
[MB-4225] capture per segment

### PR DESCRIPTION
**Pull request by pearl-truss**
## Description

This PR adds segments to the EDI to capture the phone number of the transportation office 

## Reviewer Notes

1. There are 2 types of phone number that a transportation office can have: fax or voice. This code pulls out the first voice number found. Should it instead choose a different number? I looked in the front-end and saw that the first number was being selected there as well.

2. Are the validations and default values for the PER segment as expected? I pulled from https://docs.google.com/document/d/1t4nokaYMoaQ2uBzhmxy--S3HlTTju4QhZTPIjc-Qcgs/edit

## Setup

1. Run your server `make server_run`.
2. Create a payment request with a Domestic Linehaul service item. Create a file called `create_payment_request.json` in the project root and paste in the JSON payload provided below:

```json
{
  "body": {
    "isFinal": false,
    "moveTaskOrderID": "9c7b255c-2981-4bf8-839f-61c7458e2b4d",
    "serviceItems": [
      {
        "id": "94bc8b44-fefe-469f-83a0-39b1e31116fb"
      },
      {
        "id": "eee4b555-2475-4e67-a5b8-102f28d950f8"
      },
      {
        "id": "6431e3e2-4ee4-41b5-b226-393f9133eb6c"
      },
      {
        "id": "fd6741a5-a92c-44d5-8303-1d7f5e60afbf"
      }
    ]
  }
}
```

2. Use the Prime API Client to create the payment request and return the payment request number by running the command below:

```sh
prime-api-client --insecure create-payment-request --filename  create_payment_request.json| jq '.paymentRequestNumber'
```
3. Use the CLI utility to generate the EDI858 and view it in your terminal. Remember to pass the payment request number of the payment request you just created.

```sh
go run ./cmd/generate-payment-request-edi/ --payment-request-number [$paymentRequestNumber]
```
### Expected Output
```txt
ISA*00*0084182369*00*_   _*ZZ*GOVDPIBS*12*8004171844*20201014*1803*U*00401*100001272*0*T*|
GS*SI*MYMOVE*8004171844*20201014*1803*100001251*X*004010
ST*858*0001
BX*00*J*PP*2103-4844*TRUS**4
N9*CN*2103-4844-4**
N9*CT*TRUSS_TEST**
N9*1W*Leo, Spacemen**
N9*ML*E_1**
N9*3L*ARMY**
G62*10*20201014*0*
G62*76*20201013*0*
G62*86*20200316*0*
N1*ST**10*CNNQ
N3*Fort Gordon*
N4*Augusta*GA*30813*United States**
PER*CN**TE*706-791-4184  <--- NEW LINE
N1*SF*4lNPgn8mpO*10*LKNQ
N3*987 Other Avenue*P.O. Box 1234
N4*Des Moines*IA*50309*US**
PER*CN**TE*(510) 555-5555 <--- NEW LINE
FA1*DY
FA2*TA*F8E1
HL*1**|
N9*PO*2103-4844-9bbf7ddd**
L5*1*DDP*TBD*D**
L0*1***1349.000*B******L
L3*1349.000*B*10835
HL*2**|
N9*PO*2103-4844-d5d26b84**
L5*2*DSH*TBD*D**
L0*2*373.000*DM*1349.000*B******L
L3*1349.000*B*457053
HL*3**|
N9*PO*2103-4844-97fc8e42**
L5*3*DLH*TBD*D**
L0*3*373.000*DM*1349.000*B******L
L3*1349.000*B*1221717
HL*4**|
N9*PO*2103-4844-c96bdb89**
L5*4*FSC*TBD*D**
L0*4*373.000*DM*1349.000*B******L
L3*1349.000*B*362
SE*41*0001
GE*1*100001251
IEA*1*100001272
```
## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4225) for this change

### CODE REVIEW

**Review by gidjin — COMMENTED** [review]
Changes look good, everything worked locally. I left a few comments about the validations rules around the PER section that seemed different than what was in the code.

**Comment by pearl-truss** [comment]
Thanks for reviewing @gidjin ! Comments are resolved and I think I have a better on how to parse the 858 documentation

File: pkg/edi/segment/per.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		"PER",
+		s.ContactFunctionCode,
+		s.Name,
+		s.CommunicationNumberQualifier,
+		s.CommunicationNumber,
+	}
+}
+
+// Parse parses an X12 string that's split into an array into the PER struct
+func (s *PER) Parse(parts []string) error {
```
Comment (jacquelineIO) [inline pkg/edi/segment/per.go:28]: @pearl-truss  is the `Parse` function being tested?

**Review by jacquelineIO — APPROVED** [review]
I would like to see a unit test for the Parse function.
But overall, changes look good!

File: pkg/edi/segment/per.go (hunk 1 above)
Comment (pearl-truss) [reply pkg/edi/segment/per.go:28]: Each segment like l5, n9, etc have a parse function and but from checking just now none have unit test for it. I can of course add test here for this segment if necessary

Comment (jacquelineIO) [reply pkg/edi/segment/per.go:28]: @pearl-truss  yes please, we should be testing these.

**Comment by pearl-truss** [comment]
> I would like to see a unit test for the Parse function.
> But overall, changes look good!

Thanks for reviewing this @jacquelineIO ! I just added a test to check that the struct is being parsed correctly

**Pull request merged**